### DISCUSSION TITLE
Use chezmoi for temporary setups (only for that one SSH connection)?

### DISCUSSION THREAD

**Discussion by old-pigeon** [twpayne/chezmoi#3675] [Q&A] [answered]
Hey, chezmoi looks amazing!

I am about to try it, but at work I have this scenario: multiple Linux servers where multiple people share the same user. Yes, I know it's bad practice but I am not in a position to change this.

This means that I can't simply execute `chezmoi apply` because if I do, I destroy the experience for all other persons at my company using the same user on those machines.

I though about doing this instead, but I am not sure if it's a good idea:

1. Use `chez init` with a custom source directory containing a personal identifier of mine to absolutely avoid messing with someones data, i. e. `~/.chezmoi-totallyuniquedir/`
2. Execute `export HOME=/home/myuser/.chezmoi-totallyuniquedir`
3. Execute `zsh` (or whatever shell I want to use)
4. Execute `chez apply` with that custom `HOME` path

This would effectively make that unique directory my temporary home, meaning I have all my stuff now in the shell executed by step 3. It also makes sure that all config files etc. are properly picked up by any tools I'd use.

I guess it would work in most cases, but modifying `HOME` feels like it will break things/tools in unforeseeable ways.

Is there a better way to use chezmoi for this kind of temporary setup or is the above suggestion the best I can get away with?

I am fine with my data staying on the server as long as it doesn't interfere with any other person (I don't want to break their flow).

Thanks for your help!

**Comment by twpayne** [comment, marked as answer]
> I guess it would work in most cases, but modifying `HOME` feels like it will break things/tools in unforeseeable ways.

Yes, modifying `HOME` will break a lot of things. I have no idea how many. As a first guess: it will break everything.

> Is there a better way to use chezmoi for this kind of temporary setup or is the above suggestion the best I can get away with?

I would suggest the following:

1. Since most things are controlled by your shell and environment variables, I would create a shell script that you `source` when you log into the machine which sets environment variables and spawns the shell you want.
2. You can store this script in `$HOME/.old-pigeon.sh`, which you can manage with chezmoi. When you login you run `. .old-pigeon.sh`.
3. `.old-pigeon.sh` sets environment variables like `$XDG_CONFIG_HOME` to set your config directory. You can use this to point most applications configurations to a directory you control (via chezmoi). For example, you configure chezmoi to populate `$HOME/.config/old-pigeon/git/config` and set `$XDG_CONFIG_HOME` to be `$HOME/.config/old-pigeon` in `$HOME/.old-pigeon.sh`.
4. As a bonus, running `. .old-pigeon.sh` can finish with an `exec old-pigeons-favorite-shell` (maybe `exec zsh`) which will replace the shell with your preferred one.

Hope this helps. chezmoi is probably useful for running botnets, but I hope that's not your usecase.

**Reply by old-pigeon** [reply]
Thanks, I'll try this!

If I ever come up to build a botnet, I'll think of chezmoi. But for now be assured that it's just my company which "grew historically" and in a rush SSH keys were added to existing users like `admin` instead of creating new ones and no one has the time to fix it 😅

**Reply by twpayne** [reply]
For future reference, #1410 is also relevant.
